lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1496,7 +1482,7 @@ static void checkTypeAccessibilityImpl(
   if (!typeAccessScope.hasValue())
     return;
 
-  if (!*typeAccessScope || *typeAccessScope == contextAccessScope)
+  if (typeAccessScope->isPublic() || *typeAccessScope == contextAccessScope)
```

Comment:
Then it probably should break something, but I coudn't come up with a failing scenario and all tests are passing locally. Would be great to have a test case for this.